### PR TITLE
Planner hook fix

### PR DESCRIPTION
**Pull request by orduno**
## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #304 |
| Primary OS tested on | (Ubuntu) |
| Primary platform tested on | (gazebo turtlebot) |

--- 

## Description of contribution in a few bullet points

The artifact observed at the end of the path is originated from rounding errors when translating the goal and start poses from world continuous space to the discretized map representation.

The figure below shows an example. To the left we have a path generated with a 'hook' artifact on one end, to the right is a path (connecting the same start and end poses) but without artifact.

both

### CODE REVIEW

File: nav2_smart_planner/src/smart_planner.cpp [hunk 1]
```diff
@@ -201,7 +202,7 @@ SmartPlanner::makePlan(
   wx = goal.position.x;
   wy = goal.position.y;
 
-  if (worldToMap(wx, wy, mx, my)) {
+  if (!worldToMap(wx, wy, mx, my)) {
     if (tolerance <= 0.0) {
```
Comment (crdelsey) [inline nav2_smart_planner/src/smart_planner.cpp]: Does the tolerance value actually matter here? Seems we should error out and fail regardless of the tolerance setting here.

File: nav2_smart_planner/src/smart_planner.cpp [hunk 2]
```diff
@@ -269,6 +269,25 @@ SmartPlanner::makePlan(
   return !plan.poses.empty();
 }
 
+void
+SmartPlanner::smoothApproachToGoal(
```
Comment (crdelsey) [inline nav2_smart_planner/src/smart_planner.cpp]: I think a comment explaining what this function is doing would be helpful.

File: nav2_smart_planner/src/smart_planner.cpp [hunk 3]
```diff
@@ -269,6 +269,25 @@ SmartPlanner::makePlan(
...
+  const geometry_msgs::msg::Pose & best_pose,
```
Comment (mkhansen-intel) [inline nav2_smart_planner/src/smart_planner.cpp]: In the header file the Pose is called "goal", here you're calling it "best_pose". I'm confused? I think goal_pose is the most clear

File: nav2_smart_planner/src/smart_planner.cpp (hunk 1 above)
Comment (orduno) [reply nav2_smart_planner/src/smart_planner.cpp]: Carryover code from ROS1, agree, needs to be changed.

File: nav2_smart_planner/src/smart_planner.cpp (hunk 2 above)
Comment (orduno) [reply nav2_smart_planner/src/smart_planner.cpp]: Will add comments.

File: nav2_smart_planner/src/smart_planner.cpp (hunk 3 above)
Comment (orduno) [reply nav2_smart_planner/src/smart_planner.cpp]: I'll fix this.

**Comment by orduno** [comment]
Thanks for your feedback, I've addressed all and it's ready to merge. I'll go ahead and do it.

**Pull request merged**